### PR TITLE
DNS check

### PR DESCRIPTION
**Pull request by alexandresobolevski**
The browser can visit the url
screen shot 2017-04-11 at 12 23 48 am

but the fetch from the electron app or from postman fails with `ERR_INSECURE_RESPONSE`
@chriddyp any ideas why?
screen shot 2017-04-11 at 12 23 56 am

I tried with and without the commented out code.
```
    return fetch(url, {
        method: 'GET'
        // credentials: 'include',
        // headers: {
        //     'Accept': 'application/json',
        //     'Content-Type': 'application/json'
        // }
    })
    .then(res => {
        console.log('fetched');
        return true;
    })
    .catch((e) => {
        console.log('failed');
        return false;
    });

```

### CODE REVIEW

**Comment by chriddyp** [comment]
It looks like this answer explains why: http://stackoverflow.com/questions/31058764/determine-if-ajax-call-failed-due-to-insecure-response-or-connection-refused

So it looks like that's normal browser behaviour unfortunately 😿 

We'll probably just need to set an initial `status: 'unresolved'` variable and then resolve it if the `catch` (`status: error`) and the `then` (`status: success`)

**Comment by alexandresobolevski** [comment]
Ah damn, no way around?

> We'll probably just need to set an initial status: 'unresolved' variable and then resolve it if the catch (status: error) and the then (status: success)

Not sure what you mean - Can you take a loot at the PR and see what you want me to change?

File: app/components/Settings/Settings.react.js [hunk 1]
```diff
@@ -262,6 +285,23 @@ class Settings extends Component {
...
         const connectorUrl = this.state.urls.https || this.state.urls.http;
 
+        let httpsServerIsOK = false;
+        if (!STARTED_AT && this.state.urls.https) {
+            STARTED_AT = new Date();
+            checkDNS = setInterval(() => {
+                console.log('trying');
+                if (isStatusOK(`${this.state.urls.https}/ping`)) {
+                    httpsServerIsOK = true;
+                }
```
Comment (chriddyp) [inline app/components/Settings/Settings.react.js]: `isStatusOK` returns a promise, not a boolean.

This should be like:
```
fetch(`${this.state.urls.https}/ping`).then(() => this.setState({httpsServerIsOK: true});
```

File: app/components/Settings/Settings.react.js [hunk 2]
```diff
@@ -306,11 +346,22 @@ class Settings extends Component {
...
-                                    <div className={styles.url}>
-                                        {`${this.state.urls[url]}`}
+                                    <div>
+                                        <div className={styles.url}>
+                                            {`${this.state.urls[url]}`}
+                                        </div>
                                     </div>
                                 );
                             })}
+                            {httpsServerIsOK ? (
```
Comment (chriddyp) [inline app/components/Settings/Settings.react.js]: and then this would just be like `this.state.httpsServerIsOK`

**Comment by alexandresobolevski** [comment]
@chriddyp But the problem is that `fetch(`${this.state.urls.https}/ping`)` is just never resolved so `status: 'unresolved' is always the case.

**Comment by chriddyp** [comment]
@alexandresobolevski - I don't follow. Won't it get resolved in 5 minutes when the DNS resolves? 
image

**Pull request merged**